### ISSUE TITLE
Key error occurring in Flocker cinder module as cinder_client.version returns "2.0" and not "2"

### ISSUE THREAD

**Issue by garthmbooth** [ClusterHQ/flocker#2841]
I brought up a Flocker cluster on an OpenStack Mitaka devstack and hit a key exception in the flocker cinder.py module. This issue occurs because cinder_client.version returns 2.0 and there is no 2.0 key in the CINDER_V1_ADAPTERS dict (only “2” and “1”). After modifying the CINDER_V1_ADAPTERS dict to include “2.0” instead of “2”, this exception went away. However, I still needed to apply the floating IP fix in order to have the dataset-agents converge.

Here's the exception details:

```
{"task_uuid": "4d35573d-c8cb-42f6-96fd-19ba01bb1d17", "error": true, "timestamp": 1466094273.831534, "message": "main function encountered error\nTraceback (most recent call           last):\n  File \"/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py\", line 84, in flocker_dataset_agent_main\n    options=options,\n  File \"/opt/flocke          r/local/lib/python2.7/site-packages/flocker/common/script.py\", line 295, in main\n    self._react(run_and_log, [], _reactor=self._reactor)\n  File \"/opt/flocker/local/lib/          python2.7/site-packages/twisted/internet/task.py\", line 908, in react\n    finished = main(_reactor, *argv)\n  File \"/opt/flocker/local/lib/python2.7/site-packages/flocker          /common/script.py\", line 283, in run_and_log\n    d = maybeDeferred(self.script.main, reactor, options)\n--- <exception caught here> ---\n  File \"/opt/flocker/local/lib/py          thon2.7/site-packages/twisted/internet/defer.py\", line 150, in maybeDeferred\n    result = f(*args, **kw)\n  File \"/opt/flocker/local/lib/python2.7/site-packages/flocker/n          ode/script.py\", line 295, in main\n    self.service_factory(reactor, options)\n  File \"/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py\", line 594, i          n get_service\n    api = agent_service.get_api()\n  File \"/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py\", line 533, in get_api\n    cluster_id\n  F          ile \"/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py\", line 417, in get_api\n    return backend.api_factory(**api_args)\n  File \"/opt/flocker/local/          lib/python2.7/site-packages/flocker/node/agents/cinder.py\", line 992, in cinder_from_configuration\n    cinder_client.version\nexceptions.KeyError: '2.0'\n", "message_type"          : "twisted:log", "task_level": [1]}
```

```
4d35573d-c8cb-42f6-96fd-19ba01bb1d17 -> /1
2016-06-16 16:24:33.831534Z
  message_type: 'twisted:log'
  error: True
  message: 'main function encountered error
         |  Traceback (most recent call           last):
         |    File "/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py", line 84, in flocker_dataset_agent_main
         |      options=options,
         |    File "/opt/flocke          r/local/lib/python2.7/site-packages/flocker/common/script.py", line 295, in main
         |      self._react(run_and_log, [], _reactor=self._reactor)
         |    File "/opt/flocker/local/lib/          python2.7/site-packages/twisted/internet/task.py", line 908, in react
         |      finished = main(_reactor, *argv)
         |    File "/opt/flocker/local/lib/python2.7/site-packages/flocker          /common/script.py", line 283, in run_and_log
         |      d = maybeDeferred(self.script.main, reactor, options)
         |  --- <exception caught here> ---
         |    File "/opt/flocker/local/lib/py          thon2.7/site-packages/twisted/internet/defer.py", line 150, in maybeDeferred
         |      result = f(*args, **kw)
         |    File "/opt/flocker/local/lib/python2.7/site-packages/flocker/n          ode/script.py", line 295, in main
         |      self.service_factory(reactor, options)
         |    File "/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py", line 594, i          n get_service
         |      api = agent_service.get_api()
         |    File "/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py", line 533, in get_api
         |      cluster_id
         |    F          ile "/opt/flocker/local/lib/python2.7/site-packages/flocker/node/script.py", line 417, in get_api
         |      return backend.api_factory(**api_args)
         |    File "/opt/flocker/local/          lib/python2.7/site-packages/flocker/node/agents/cinder.py", line 992, in cinder_from_configuration
         |      cinder_client.version
         |  exceptions.KeyError: \'2.0\'
         |  '


```

**Comment by wallnerryan** [comment]
thanks @garthmbooth will look into this

**Comment by wallrj** [comment]
I formatted the log entry in  the description.

**Comment by wallrj** [comment]
Thanks @garthmbooth 

This is my fault. 
We updated all our dependencies in:
 * https://github.com/ClusterHQ/flocker/pull/2792/files#diff-b4ef698db8ca845e5845c4618278f29aL55
 
Switching from `python-cinderclient==1.6.0`  to `python-cinderclient==1.7.1` pulled in this change:
 * https://github.com/openstack/python-cinderclient/commit/3f75b48f067abae846655e48ce8f96fda79a3ae8#diff-159aff614e80016c8a62612136f5ef28

And our Rackspace based Cinder tests never exercise the v2 driver code.

We'll fix this in time for our 1.14.0 release  on 2016-07-12.

**Comment by wallrj** [comment]
I created a Jira issue for this: https://clusterhq.atlassian.net/browse/FLOC-4461
